### PR TITLE
Devtools tab

### PR DESCRIPTION
**Pull request by Gusted**

### CODE REVIEW

File: src/ui/devtools/components/body.tsx [hunk 1]
```diff
@@ -30,11 +30,23 @@ function Body({data, actions}: BodyProps) {
...
         }
+        node.addEventListener('keydown', (e) => {
+            if (e.key === 'Tab') {
+                e.preventDefault();
+                const start = node.selectionStart;
+                const end = node.selectionEnd;
+                const before = node.value.substring(0, start);
+                const after = node.value.substring(end);
+                node.focus();
+                node.value = `${before}\t${after}`;
```
Comment (alexanderby) [inline src/ui/devtools/components/body.tsx:46]: Let's replace `\t` with 4 spaces.

File: src/ui/devtools/style.less [hunk 2]
```diff
@@ -65,6 +65,7 @@ header {
     outline: none;
     padding: @size-control-inner / 2 - @size-text-normal / 2;
     resize: none;
+    tab-size: 4;
```
Comment (alexanderby) [inline src/ui/devtools/style.less:68]: We are not using Tab symbol in config, so there should be no need for this property.

File: src/ui/devtools/components/body.tsx (hunk 1 above)
Comment (Gusted) [reply src/ui/devtools/components/body.tsx:46]: For some reason it would just add 1 space... Render body is resetting it to 1 space.......

Comment (Gusted) [reply src/ui/devtools/components/body.tsx:46]: image
Bad idea to use 4 spaces

File: src/ui/devtools/components/body.tsx [hunk 3]
```diff
@@ -30,11 +30,23 @@ function Body({data, actions}: BodyProps) {
...
         }
+        node.addEventListener('keydown', (e) => {
+            if (e.key === 'Tab') {
+                e.preventDefault();
+                const start = node.selectionStart;
+                const end = node.selectionEnd;
+                const before = node.value.substring(0, start);
+                const after = node.value.substring(end);
+                node.focus();
+                node.value = `${before}  ${after}`;
```
Comment (alexanderby) [inline src/ui/devtools/components/body.tsx:46]: Make it 4 spaces.

File: src/ui/devtools/components/body.tsx [hunk 4]
```diff
@@ -30,11 +30,23 @@ function Body({data, actions}: BodyProps) {
...
+                node.setSelectionRange(start + 1, start + 1);
```
Comment (alexanderby) [inline src/ui/devtools/components/body.tsx:47]: Fix the cursor position, play with the selection range to place it after the inserted 4 spaces.

**Pull request merged**